app/src/main/java/at/roteskreuz/stopcorona/screens/debug/exposure_notifications/DebugExposureNotificationsViewModel.kt

```diff
@@ -164,9 +165,17 @@ class DebugExposureNotificationsViewModel(
         exposureNotificationClient.temporaryExposureKeyHistory
             .addOnSuccessListener {
                 Timber.d("got the list of Temporary Exposure Keys $it")
-                exposureNotificationsTextSubject.onNext("got the list of Temp" +
-                    "oraryExposureKeys $it")
-                lastTemporaryExposureKeysSubject.onNext(it)
+                exposureNotificationsTextSubject.onNext(
```

Comment:
Why is this complexity needed here? we could debug upload with the non-random keys even...

Reply:
Because all the temporary exposure keys associated to a rollingStartIntervalNumber have the same UUID. There is a second debug method `uploadKeys` which performs an upload with the keys stored in the `lastTemporaryExposureKeysSubject` subject, in order to perform this update we need a properly structured data.